### PR TITLE
Update docrep

### LINKED ISSUE #308: Docrep has GPL license
Hey @rabernat. Our legal team was recently reviewing our dependency stack in preparation for us open sourcing some of our code and we noticed that `docrep` has as a [GPL license](https://github.com/Chilipp/docrep/blob/master/LICENSE). Since this is dynamically imported here:
https://github.com/xgcm/xgcm/blob/34a77de7894a30cb80abcdd9c20571dbe54988ad/xgcm/grid.py#L10
This implies that xgcm is really available under a GPL license rather than the claimed MIT license. 

Since this import seems rather minor, would it possible to remove this runtime dependency on docrep? Currently it's not really possible for us to use xgcm. I think it's still okay of the docs need GPL code to be built, as long as the sdist on PyPI doesn't include GPL-derived stuff.

As an aside, we joined in on similar issues with necdf4 and cftime (see e.g. https://github.com/Unidata/netcdf4-python/issues/1073).

### PR DESCRIPTION
**Pull request by TomNicholas**
An attempt to fix #308.

This PR actually just completely removes the need for docrep at all - @jbusecke and I attempted to [update docrep](https://docrep.readthedocs.io/en/latest/#changed-in-0-3) to v0.3 but kept getting errors we didn't understand (in particular this one https://github.com/xgcm/xgcm/issues/210).

So we used the nuclear option to remove docrep by literally just copying and pasting the duplicated docstrings... :confused: 

That's annoying to have to do, but in fact we should probably wait until the [grid ufunc refactor](https://github.com/xgcm/xgcm/projects/10) is complete before trying to fix this situation/reinstate docrep, because that refactor will change how these docstrings all relate to one another anyway. Seems more important right now that our users don't have license problems (@nbren12).

 - [x] Closes #308
 - [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: xgcm/grid.py [hunk 1]
```diff
@@ -1772,16 +1989,43 @@ def interp_2d_vector(self, vector, **kwargs):
...
+            * 'fill':  Set values outside the array boundary to fill_value
+              (i.e. a Dirichlet boundary condition.)
+            * 'extend': Set values outside the array to the nearest array
+              value. (i.e. a limited form of Neumann boundary condition.)
+
+            Optionally a dict with separate values for each axis can be passed (see example)
+        fill_value : {float, dict}, optional
+            The value to use in the boundary condition with `boundary='fill'`.
+            Optionally a dict with seperate values for each axis can be passed (see example)
+        vector_partner : dict, optional
```
Comment (jbusecke) [inline xgcm/grid.py:2020]: I believe the 'vector_partner' input is only valid for the methods ending in `_vector`, and should be removed from all others.

File: xgcm/grid.py [hunk 2]
```diff
@@ -592,8 +583,26 @@ def interp(
...
 
+            * None:  Do not apply any boundary conditions. Raise an error if
+              boundary conditions are required for the operation.
+            * 'fill':  Set values outside the array boundary to fill_value
+              (i.e. a Dirichlet boundary condition.)
+            * 'extend': Set values outside the array to the nearest array
+              value. (i.e. a limited form of Neumann boundary condition.)
+        fill_value : float, optional
+            The value to use in the boundary condition with `boundary='fill'`.
+        vector_partner : dict, optional
```
Comment (jbusecke) [inline xgcm/grid.py:602]: remove (see comment below)

File: xgcm/grid.py [hunk 3]
```diff
@@ -1822,17 +2065,44 @@ def integrate(self, da, axis, **kwargs):
...
-            Name of the axis on which to act
-        %(grid_func.parameters)s
+        axis : str or list or tuple
+            Name of the axis on which to act. Multiple axes can be passed as list or
+            tuple (e.g. ``['X', 'Y']``). Functions will be executed over each axis in the
+            given order.
+        to : str or dict, optional
+            The direction in which to shift the array (can be ['center','left','right','inner','outer']).
+            If not specified, default will be used.
+            Optionally a dict with seperate values for each axis can be passed (see example)
```
Comment (jbusecke) [inline xgcm/grid.py:2082]: The `(see example)` does not apply?

**Review by jbusecke — CHANGES_REQUESTED** [review]
Just minor stuff in addition to removing `vector_partner` from most of the docstrings. Otherwise I support this nuclear option for now.

File: xgcm/grid.py (hunk 1 above)
Comment (jbusecke) [reply xgcm/grid.py:2020]: Sorry this is actually false. This only applies to certain methods, namely: `cumint` and `cumsum`.

**Comment by jbusecke** [comment]
Could you rephrase

```
metric_weighted : str or tuple of str or dict, optional
            If an axis or list of axes is specified,
            the appropriate grid metrics will be used to determined the weight for interpolation.
            E.g. if passing `metric_weighted=['X', 'Y']`, values will be weighted by horizontal area.
            If `False` (default), the points will be weighted equally.
            Optionally a dict with seperate values for each axis can be passed (see example)
```

to 

```
metric_weighted : str or tuple of str or dict, optional
            Optionally use metrics to multiply/divide with appropriate metrics before/after the operation.
            E.g. if passing `metric_weighted=['X', 'Y']`, values will be weighted by horizontal area.
            If `False` (default), the points will be weighted equally.
            Optionally a dict with seperate values for each axis can be passed.
```

Otherwise this looks good to me. Thanks a lot @TomNicholas.

**Review by jbusecke — APPROVED** [review]

**Comment by rabernat** [comment]
Thanks so much for taking care of this chore Tom! 👏

**Comment by nbren12** [comment]
Thanks! Doing God's work here!
